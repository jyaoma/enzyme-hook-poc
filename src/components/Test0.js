import React from 'react';

// const Test0 = () => <div>Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Maecenas porttitor congue massa. Fusce posuere, magna sed pulvinar ultricies, purus lectus malesuada libero, sit amet commodo magna eros quis urna.
// Nunc viverra imperdiet enim. Fusce est. Vivamus a tellus.
// Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Proin pharetra nonummy pede. Mauris et orci.
// Aenean nec lorem. In porttitor. Donec laoreet nonummy augue.
// Suspendisse dui purus, scelerisque at, vulputate vitae, pretium mattis, nunc. Mauris eget neque at sem venenatis eleifend. Ut nonummy.</div>

class Test0 extends React.Component {
  constructor(props, context) {
    super(props, context);

    this.state = {
      toggleState: false
    };
  }

  render() {
    return <div>
      <input type="checkbox" onChange={() => this.setState({toggleState: !this.state.toggleState})} checked={this.state.toggleState} />
      <div>{this.state.toggleState ? 'True' : 'False'}</div>
    </div>
  }
}

export default Test0;