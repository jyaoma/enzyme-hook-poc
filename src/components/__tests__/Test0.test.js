import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import Test0 from '../Test0';

describe('Test0', () => {
  it('should render correctly', () => {
    const rendered = renderer.create(<Test0/>);
    expect(rendered.toJSON()).toMatchSnapshot();
  });

  it('should render again', () => {
    const wrapper = shallow(<Test0 />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should be able to see props', () => {
    const tree = shallow(<Test0 testProp="oof" />);
    // console.warn(tree.props().testProp);
    // expect(tree.props().testProp).toEqual('oof');
    console.warn(tree.props());
  });
});