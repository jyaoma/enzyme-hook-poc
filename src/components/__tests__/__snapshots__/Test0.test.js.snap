// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test0 should render again 1`] = `
<div>
  <input
    checked={false}
    onChange={[Function]}
    type="checkbox"
  />
  <div>
    False
  </div>
</div>
`;

exports[`Test0 should render correctly 1`] = `
<div>
  <input
    checked={false}
    onChange={[Function]}
    type="checkbox"
  />
  <div>
    False
  </div>
</div>
`;
